Update source functions - include variable scoping [DOC-1065].md (#7513)
segmentio/segment-docs: src/connections/functions/source-functions.md

> warning ""
> When you use the `set()` method, you won't see events in the Source Debugger. Segment only sends events to connected warehouses.

### Variable scoping

> When declaring settings variables, make sure to declare them in the function handler rather than globally in your Function. This prevents you leaking the settings values across other function instances. 

Reviewer: ```suggestion
Declare settings variables in the function handler, rather than globally in your function. This prevents you from leaking the settings values across other function instances. 
```
